src/AddressRegistry.Api.Legacy/AddressMatch/Matching/StreetNameMatcher.cs

```diff
@@ -89,73 +85,112 @@ protected override AddressMatchBuilder DoMatchInternal(AddressMatchBuilder resul
-            {
-                municipalityWrapper.AddStreetNames(_streetNameService.GetStreetNamesByKadStreet(results.Query.KadStreetNameCode, municipalityWrapper.NisCode));
-                //_telemetry.TrackStraatnaamMatch(gemeente.Naam, results.Query.Straatnaam, results.Query.KadStraatcode, results.Query.RrStraatcode, gemeente.Count(), (int)StraatnaamMatchLevel.KadStraatCode);
-            }
+            // TODO: Can't we skip this entire bit if there is no KadStreetNameCode query?
 
-            if (!string.IsNullOrEmpty(results.Query.StreetName) && !results.AllStreetNames().Any(w => w.StreetName.NameDutch.EqIgnoreCase(results.Query.StreetName)))
+            foreach (var municipalityWrapper in results)
+                municipalityWrapper
+                    .AddStreetNames(
+                        _streetNameService
+                            .GetStreetNamesByKadStreet(
+                                results.Query.KadStreetNameCode,
+                                municipalityWrapper.NisCode));
+
+            // TODO: Shouldnt we be matching on w.StreetName.GetDefaultName()?
+            if (!string.IsNullOrEmpty(results.Query.StreetName) &&
+                !results.AllStreetNames().Any(w => w.StreetName.NameDutch != null && w.StreetName.NameDutch.EqIgnoreCase(results.Query.StreetName)))
                 _warnings.AddWarning("7", "Geen overeenkomst tussen 'KadStraatcode' en 'Straatnaam'.");
         }
 
         private void FindStreetNamesByRrStreetCode(AddressMatchBuilder results)
         {
-            var streetName = _streetNameService.GetStreetNameByRrStreet(results.Query.RrStreetCode, results.Query.PostalCode);
+            var streetName = _streetNameService
+                .GetStreetNameByRrStreet(
+                    results.Query.RrStreetCode,
+                    results.Query.PostalCode);
+
             if (streetName != null)
-            {
-                results.Where(g => g.PostalCode == results.Query.PostalCode).ToList().ForEach(g => g.AddStreetName(streetName));
-                //_telemetry.TrackStraatnaamMatch(results.Where(g => g.Postcode == results.Query.Postcode).FirstOrDefault()?.Naam, results.Query.Straatnaam, results.Query.KadStraatcode, results.Query.RrStraatcode, 1, (int)StraatnaamMatchLevel.RrStraatCode);
-            }
+                results
+                    .Where(g => g.PostalCode == results.Query.PostalCode)
+                    .ToList()
+                    .ForEach(g => g.AddStreetName(streetName));
 
-            if (!string.IsNullOrEmpty(results.Query.StreetName) && !results.AllStreetNames().Any(w => w.StreetName.NameDutch.EqIgnoreCase(results.Query.StreetName)))
+            // TODO: Shouldnt we be matching on w.StreetName.GetDefaultName()?
+            if (!string.IsNullOrEmpty(results.Query.StreetName) && !results.AllStreetNames().Any(w => w.StreetName.NameDutch != null && w.StreetName.NameDutch.EqIgnoreCase(results.Query.StreetName)))
```

Comment:
Shouldnt we be matching on `w.StreetName.GetDefaultName()`?

Reply:
Same comment as with municipality. Only when we look through all names.